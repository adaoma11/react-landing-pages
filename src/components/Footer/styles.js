import styled, { css } from 'styled-components';

export const Footer = styled.footer`
  ${({ theme, $hasBg }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    height: 150px;
    background-color: ${$hasBg ? theme.colors.primary : theme.colors.white};
  `}
`;
