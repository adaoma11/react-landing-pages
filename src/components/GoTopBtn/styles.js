import styled, { css } from 'styled-components';

export const Link = styled.a`
  ${({ theme }) => css`
    display: flex;
    width: 40px;
    height: 40px;
    position: fixed;
    bottom: 2rem;
    right: 2rem;
    color: ${theme.colors.white};
    background-color: ${theme.colors.primary};
    border-radius: 5px;
    opacity: 0.8;
  `}
`;
