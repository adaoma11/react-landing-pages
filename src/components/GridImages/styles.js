import styled from 'styled-components';

export const GridContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  width: 100%;
  gap: 20px;
  margin-top: 50px;
`;

export const GridItem = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 0.5s ease-in-out;

  &:hover {
    transform: scale(1.2) rotate(5deg);
  }
`;
