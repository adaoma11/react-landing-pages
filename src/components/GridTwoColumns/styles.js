import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 1fr;
    align-items: center;
    gap: 30px;

    @media ${theme.medias.medium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const TextContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 30px;
`;

export const Image = styled.div`
  ${({ $imgSrc }) => css`
    display: block;
    min-height: 500px;
    background-image: url(${$imgSrc});
    background-repeat: no-repeat;
    background-position: center;
    background-size: 60%;
  `}
`;
