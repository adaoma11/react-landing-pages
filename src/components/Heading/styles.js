import styled, { css } from 'styled-components';

export const Title = styled.h1`
  ${({ theme, $dark, size, $upperCase }) => css`
    color: ${$dark ? theme.colors.primary : theme.colors.white};

    font-size: ${theme.fonts.sizes[size]};

    ${size == 'xlarge' &&
    css`
      @media ${theme.medias.small} {
        font-size: ${theme.fonts.sizes.large};
      }
    `}

    text-transform: ${$upperCase ? 'uppercase' : 'none'}
  `}
`;
