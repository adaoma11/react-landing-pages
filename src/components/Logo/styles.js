import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.heading};
    font-size: ${theme.fonts.sizes.large};
    font-weight: bold;
    color: ${theme.colors.primary};
    text-decoration: none;

    > img {
      min-height: 40px;
      max-height: 50px;
      max-width: 150px;
      min-width: 40px;
    }
  `}
`;
