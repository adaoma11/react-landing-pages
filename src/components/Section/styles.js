import styled, { css } from 'styled-components';

export const Container = styled.section`
  ${({ theme, $hasBg }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    min-height: 100vh;
    background-color: ${$hasBg ? theme.colors.primary : theme.colors.white};
  `}
`;

export const Article = styled.article`
  max-width: 1200px;
  width: 100%;
  padding: 30px;
`;
