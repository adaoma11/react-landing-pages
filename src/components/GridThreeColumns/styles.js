import styled from 'styled-components';
import { Title as GridItemTitle } from '../Heading/styles';

export const GridContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  counter-reset: grid-counter;
  width: 100%;
  gap: 50px;
`;

export const GridItem = styled.div`
  margin-top: 80px;

  ${GridItemTitle} {
    position: relative;
    left: 4rem;
    margin-bottom: 30px;
  }

  ${GridItemTitle}::before {
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 5rem;
    top: -2rem;
    left: -4rem;
    transform: rotate(7deg);
  }
`;
