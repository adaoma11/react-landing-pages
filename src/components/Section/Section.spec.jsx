import { renderTheme } from '../../styles/renderTheme';
import { theme } from '../../styles/theme';
import { Section } from '.';

describe('<Section />', () => {
  it('should render component with light background', () => {
    const { container } = renderTheme(<Section>Children</Section>);
    expect(container.firstChild).toHaveStyle(
      `background-color:${theme.colors.white}`,
    );
  });

  it('should render component with dark background', () => {
    const { container } = renderTheme(<Section hasBg>Children</Section>);
    expect(container.firstChild).toHaveStyle(
      `background-color:${theme.colors.primary}`,
    );
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<Section>Children</Section>);
    expect(container).toMatchSnapshot();
  });
});
