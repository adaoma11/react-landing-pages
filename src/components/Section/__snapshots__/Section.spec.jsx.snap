// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> should match snapshot 1`] = `
.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  min-height: 100vh;
  background-color: #FFFFFF;
}

.c1 {
  max-width: 1200px;
  padding: 30px;
}

<div>
  <section
    class="c0"
  >
    <article
      class="c1"
    >
      Children
    </article>
  </section>
</div>
`;
